File: BackEnd/src/main/java/com/PaintBackEnd/BackEnd/Shapes/ShapeDTO.java
package com.PaintBackEnd.BackEnd.Shapes;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;


@JsonIgnoreProperties(ignoreUnknown = true)
public class ShapeDTO {

    public String shapeType;
    public int id;
    public int rotation;

    // WHERE IN DRAWING AREA
    public double x_start;
    public double y_start;

    public double x_end;
    public double y_end;

    // COLOR
    public String fill;
    public String stroke;
    public String strokeWidth;

    // SQUARE // TRIANGLE
    public double side;

    // RECTANGLE
    public double width;
    public double height;

    // CIRCLE
    public double radius;

    // ELLIPSE
    public double x_radius;
    public double y_radius;

}

File: FrontEnd/paint/src/app/models/shapedto.ts
export interface ShapeDTO {
    shapeType : string,
    id : number,
    rotation : number,

    // COORDINATES
    x_start : number,
    y_start : number,

    x_end : number,
    y_end : number,

    // COLOR
    fill : string,
    stroke : string,
    strokeWidth : number,

    // SQUARE, TRIANGLE
    side? : number,

    // RECTANGLE
    width? : number,
    height? : number

    // CIRCLE
    radius? : number,

    // ELLIPSE
    radius_x? : number,
    radius_y? : number
}
